redis.ts

```diff
@@ -1116,8 +1117,13 @@ class RedisImpl implements Redis {
     return this.execStatusReply("MODULE", "UNLOAD", name);
   }
 
-  monitor() {
-    throw new Error("not supported yet");
+  async monitor() {
+    const connection = this.executor.connection.duplicate();
```

Comment:
I think it would be better not to duplicate the connection for the following reasons:

1. consistency with `subscribe()`
2. node-redis also works that way
    - https://github.com/redis/node-redis/blob/ffa7d2525c6d8bdd24faaad527a3ad0aec2cd8e9/packages/client/lib/client/index.ts#L985-L992

What do you think about this?

Reply:
I've been using ioredis historically (which also seems to be 1.5+ times more popular than node-redis on npm).

Unlike node-redis, it does duplicate the connection by default on `monitor()` which I think was rational because all other commands become unavailable then.

The _only_ positive side of using a single connection I can see is to cover the case when one would want to create redis connection _only_ to perform monitoring and nothing else.

But this comes with a cost, making it dangerous, as it easily leads to unexpected behavior, blocking execution of all other redis commands. For those who are not experienced with redis _much_ or for those coming from `ioredis`, it would become unclear, why some redis commands just get stuck and take forever to execute; in a large codebase it would be especially challenging to debug. It's also unclear whether the commands you call are accumulated and executed once you terminate the monitor (which can only be done manually), or they are just ignored entirely.

Reply:
I would advocate for the duplication of connection as it make it simpler and doesn't "break all things", as well as eliminates potential issues and unforeseen behavior.

That said, what about brining an argument (option) to the `monitor()` that could make it use the same connection?
To cover the case when one needs to have redis only for monitoring and doesn't need a regular working redis instance to use the rest of the API.

Reply:
@ArturSharapov I'm in favor of adding an option to the `monitor()` :+1: 

We are planning to support RESP3 in the future, and [it would be ideal to have an option of not duplicating a connection to `monitor()`](https://github.com/redis/redis-specifications/blob/1252427cdbc497f66a7f8550c6b5f2f35367dc92/protocol/RESP3.md#push-type).

I have two suggestions:

- Add `duplicate` option to `monitor()`.
- Make `Connection#duplicate` a private API
  - I'll implement a connection pool and would like to keep `Connection#duplicate` private until then
  - e.g.) https://github.com/denodrivers/redis/blob/ad96f2c85ce2c648a0ec6745ea7a0699c30079d2/connection.ts#L51

Reply:
Good call 👍